src/ray/raylet/node_manager.cc

```diff
@@ -649,8 +647,15 @@ void NodeManager::SubmitTask(const Task &task, const Lineage &uncommitted_lineag
       local_queues_.QueueUncreatedActorMethods({task});
     }
   } else {
-    // This is a non-actor task. Queue the task for local execution.
-    QueueTask(task);
+    // This is a non-actor task. Queue the task for a placement decision or for dispatch
+    // if the task was forwarded.
+    if (forwarded) {
+      // Check for local dependencies and enqueue as waiting or ready for dispatch.
+      TransitionPlaceableTask(task);
+    } else {
+      local_queues_.QueuePlaceableTasks({task});
```

Comment:
It might make sense to queue all new tasks as `PLACEABLE`, regardless of whether they're forwarded or not, and allow `TransitionPlaceableTask` to carry out the appropriate transition.

Reply:
yeah, I wanted to do it, but it would incur the performance penalty, especially if the `placeable` queue is large. I agree it would be cleaner, though.